crates/bevy_asset/src/reflect.rs

```diff
@@ -0,0 +1,305 @@
+use std::any::{Any, TypeId};
+
+use bevy_ecs::world::World;
+use bevy_reflect::{FromReflect, FromType, Reflect, Uuid};
+
+use crate::{Asset, Assets, Handle, HandleId, HandleUntyped};
+
+/// Type data for the [`TypeRegistry`](bevy_reflect::TypeRegistry) used to operate on reflected [`Asset`]s.
+///
+/// This type provides similar methods to [`Assets<T>`] like `get`, `add` and `remove`, but can be used in situations where you don't know which asset type `T` you want
+/// until runtime.
+///
+/// [`ReflectAsset`] can be obtained via [`TypeRegistration::data`](bevy_reflect::TypeRegistration::data) if the asset was registered using [`register_asset_reflect`](crate::AddAsset::register_asset_reflect).
+#[derive(Clone)]
+pub struct ReflectAsset {
+    type_uuid: Uuid,
+    handle_type_id: TypeId,
+    assets_resource_type_id: TypeId,
+
+    get: fn(&World, HandleUntyped) -> Option<&dyn Reflect>,
+    get_mut: fn(&mut World, HandleUntyped) -> Option<&mut dyn Reflect>,
+    get_unchecked_mut: unsafe fn(&World, HandleUntyped) -> Option<&mut dyn Reflect>,
+    add: fn(&mut World, &dyn Reflect) -> HandleUntyped,
+    set: fn(&mut World, HandleUntyped, &dyn Reflect) -> HandleUntyped,
+    len: fn(&World) -> usize,
+    ids: for<'w> fn(&'w World) -> Box<dyn Iterator<Item = HandleId> + 'w>,
+    remove: fn(&mut World, HandleUntyped) -> Option<Box<dyn Reflect>>,
+}
+
+impl ReflectAsset {
+    /// The [`bevy_reflect::TypeUuid`] of the asset
+    pub fn type_uuid(&self) -> Uuid {
+        self.type_uuid
+    }
+
+    /// The [`TypeId`] of the [`Handle<T>`] for this asset
+    pub fn handle_type_id(&self) -> TypeId {
+        self.handle_type_id
+    }
+
+    /// The [`TypeId`] of the [`Assets`] resource
+    pub fn assets_resource_type_id(&self) -> TypeId {
+        self.assets_resource_type_id
+    }
+
+    /// Equivalent of [`Assets::get`]
+    pub fn get<'w>(&self, world: &'w World, handle: HandleUntyped) -> Option<&'w dyn Reflect> {
+        (self.get)(world, handle)
+    }
+    /// Equivalent of [`Assets::get_mut`]
```

Comment:
Super Minor Nit: Spacing
```suggestion
    }

    /// Equivalent of [`Assets::get_mut`]
```